modules/reindex/src/main/java/org/elasticsearch/index/reindex/AbstractAsyncBulkByScrollAction.java

```diff
@@ -275,23 +285,27 @@ void onBulkResponse(BulkResponse response) {
                 startNormalTermination(emptyList(), emptyList());
                 return;
             }
-            startNextScroll();
+            startNextScroll(response.getItems().length);
         } catch (Throwable t) {
             finishHim(t);
         }
     }
 
-    void startNextScroll() {
+    void startNextScroll(int lastBatchSize) {
```

Comment:
Yeah - its almost certainly past time for all the package private methods to have javadocs.